cargo fails to build as root: `make install` complains "Please run `make` first"

I'm trying to install cargo from github by running the usual sequence of commands (as root).

```
git clone http://github.com/rust-lang/cargo cargo-git
cd cargo-git
./configure
make
make install
```

`make` completes successfully, ie. it returns status code 0 and doesn't generate any error messages. But then `make install` says "Please run `make` first".

Here is the output of `make -n install`

```
sudo -u "$SUDO_USER" make prepare-manifest-x86_64-unknown-linux-gnu
Please run `make` first
Makefile:182: recipe for target 'prepare-manifest-x86_64-unknown-linux-gnu' failed
make: *** [prepare-manifest-x86_64-unknown-linux-gnu] Error 1
Makefile:182: recipe for target 'prepare-manifest-x86_64-unknown-linux-gnu' failed
make: *** [prepare-manifest-x86_64-unknown-linux-gnu] Error 2
```

--- a/Makefile.in
+++ b/Makefile.in
@@ -149,10 +149,6 @@ distcheck-$(1): dist-$(1)
 $$(DISTDIR_$(1))/$$(PKG_NAME)-$(1).tar.gz: prepare-manifest-$(1)
 	tar -czvf $$@ -C $$(@D) $$(PKG_NAME)-$(1)
 
-ifeq (root user, $$(USER) $$(patsubst %,user,$$(SUDO_USER)))
-prepare-manifest-$(1):
-	@sudo -u "$$$$SUDO_USER" $$(MAKE) prepare-manifest-$(1)
-else
 prepare-manifest-$(1):
 	@[ -f $$(TARGET_$(1))/cargo$$(X) ] || echo 'Please run `make` first'
 	@[ -f $$(TARGET_$(1))/cargo$$(X) ]
@@ -172,7 +168,6 @@ prepare-manifest-$(1):
 		$$(PKGDIR_$(1))
 	mv $$(DISTDIR_$(1))/manifest-$$(PKG_NAME).in \
 		$$(PKGDIR_$(1))/lib/cargo/manifest.in
-endif
 
 install-$(1): prepare-manifest-$(1)
 	$$(PKGDIR_$(1))/install.sh \

